improver_tests/psychrometric_calculations/condensation_trails/test_CondensationTrailFormation.py

```diff
@@ -223,3 +223,147 @@ def test_find_local_vapour_pressure(
     plugin.relative_humidity = relative_humidity
     result = plugin._find_local_vapour_pressure(pressure_levels)
     np.testing.assert_allclose(result, expected_vapour_pressure)
+
+
+@pytest.mark.parametrize(
+    "engine_contrail_factors, pressure_levels, relative_humidity, expected_critical_temperatures, expected_critical_intercepts",
+    [
+        # one contrail factor, one pressure level, 1D relative humidity
+        (
+            [3e-5],
+            np.array([1e4], dtype=np.float32),
+            np.array([0, 0.5, 1], dtype=np.float32),
+            np.array([[[210.87348485, 212.74835668, 219.25]]], dtype=np.float32),
+            np.array([[-101.36159382]], dtype=np.float32),
+        ),
+        # one contrail factor, multiple pressure levels, 1D relative humidity
+        (
+            [3e-5],
+            np.array([1e5, 1e4], dtype=np.float32),
+            np.array([0, 0.5, 1], dtype=np.float32),
+            np.array(
+                [
+                    [
+                        [232.65, 235.00909317, 243.25],
+                        [210.87348485, 212.74835668, 219.25],
+                    ]
+                ],
+                dtype=np.float32,
+            ),
+            np.array([[-1126.49826644, -101.36159382]], dtype=np.float32),
+        ),
+        # one contrail factor, multiple pressure levels, 2D relative humidity
+        (
+            [3e-5],
+            np.array([1e5, 1e4], dtype=np.float32),
+            np.array([[0, 0.2, 0.4], [0.6, 0.8, 1]], dtype=np.float32),
+            np.array(
+                [
+                    [
+                        [
+                            [232.65, 233.45120317, 234.42736744],
+                            [235.68303131, 237.51159792, 243.25],
+                        ],
+                        [
+                            [210.87348485, 211.51295453, 212.28699057],
```

Comment:
We only need 7 sig figs for all the expected outputs. e.g.:
```suggestion
                            [210.8735, 211.5130, 212.2870],
```
Please can you truncate them all?